From 1e8dd1b8d917dc9bded506a518d072b9a7195e7f Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:24:24 -0400
Subject: [tools] perf/trace: Beautify rlmimit resources

Message-id: <1408462094-14194-96-git-send-email-jolsa@redhat.com>
Patchwork-id: 88114
O-Subject: [PATCH RHEL7.1 BZ1131394 095/325] perf trace: Beautify rlmimit resources
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit c045bf02e47664af67f8970a6a67065ff51acf62
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Wed Sep 4 11:52:33 2013 -0300

description
===========
On the getrlimit, setrlimit and prlimit64 syscalls.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-trace.c b/tools/perf/builtin-trace.c
index 19bc0e3..ae680b2 100644
--- a/tools/perf/builtin-trace.c
+++ b/tools/perf/builtin-trace.c
@@ -252,6 +252,13 @@ static const char *fcntl_cmds[] = {
 };
 static DEFINE_STRARRAY(fcntl_cmds);
 
+static const char *rlimit_resources[] = {
+ "CPU", "FSIZE", "DATA", "STACK", "CORE", "RSS", "NPROC", "NOFILE",
+ "MEMLOCK", "AS", "LOCKS", "SIGPENDING", "MSGQUEUE", "NICE", "RTPRIO",
+ "RTTIME",
+};
+static DEFINE_STRARRAY(rlimit_resources);
+
 static const char *sighow[] = { "BLOCK", "UNBLOCK", "SETMASK", };
 static DEFINE_STRARRAY(sighow);
 
@@ -466,6 +473,9 @@ static struct syscall_fmt {
  { .name     = "getitimer",  .errmsg = true,
    .arg_scnprintf = { [0] = SCA_STRARRAY, /* which */ },
    .arg_parm  = { [0] = &strarray__itimers, /* which */ }, },
+ { .name     = "getrlimit",  .errmsg = true,
+   .arg_scnprintf = { [0] = SCA_STRARRAY, /* resource */ },
+   .arg_parm  = { [0] = &strarray__rlimit_resources, /* resource */ }, },
  { .name     = "ioctl",     .errmsg = true,
    .arg_scnprintf = { [2] = SCA_HEX, /* arg */ }, },
  { .name     = "kill",     .errmsg = true,
@@ -498,6 +508,9 @@ static struct syscall_fmt {
  { .name     = "poll",     .errmsg = true, .timeout = true, },
  { .name     = "ppoll",     .errmsg = true, .timeout = true, },
  { .name     = "pread",     .errmsg = true, .alias = "pread64", },
+ { .name     = "prlimit64",  .errmsg = true,
+   .arg_scnprintf = { [1] = SCA_STRARRAY, /* resource */ },
+   .arg_parm  = { [1] = &strarray__rlimit_resources, /* resource */ }, },
  { .name     = "pwrite",     .errmsg = true, .alias = "pwrite64", },
  { .name     = "read",     .errmsg = true, },
  { .name     = "recvfrom",   .errmsg = true, },
@@ -514,6 +527,9 @@ static struct syscall_fmt {
  { .name     = "setitimer",  .errmsg = true,
    .arg_scnprintf = { [0] = SCA_STRARRAY, /* which */ },
    .arg_parm  = { [0] = &strarray__itimers, /* which */ }, },
+ { .name     = "setrlimit",  .errmsg = true,
+   .arg_scnprintf = { [0] = SCA_STRARRAY, /* resource */ },
+   .arg_parm  = { [0] = &strarray__rlimit_resources, /* resource */ }, },
  { .name     = "socket",     .errmsg = true,
    .arg_scnprintf = { [0] = SCA_STRARRAY, /* family */
         [1] = SCA_SK_TYPE, /* type */ },
-- 
1.7.1